test/PSRP.Tests.ps1

```diff
@@ -0,0 +1,138 @@
+            {
+                $_.FullyQualifiedErrorId | Should be "AccessDenied,PSSessionOpenFailed"
+            }
+            
+        }
+        It "Remoting from Windows to Linux with basic authentication by winrm enumerate OMI_Identify with bad username should throw exception" -Skip:($IsLinux -Or $IsOSX) {
+            $hostname = $LinuxHostName
+            $User = $badUserName
+            $Port = 5986
+            try
+            {
+                winrm enumerate http://schemas.microsoft.com/wbem/wscim/1/cim-schema/2/OMI_Identify?__cimnamespace=root/omi -r:https://${hostname}:$Port -auth:Basic -u:$User -p:$passwordString -skipcncheck -skipcacheck -encoding:utf-8
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | Should be "NativeCommandError"
+            }
+        }
+        It "Remoting from Windows to Linux with basic authentication by winrm enumerate OMI_Identify with bad password should throw exception" -Skip:($IsLinux -Or $IsOSX) {
+            $hostname = $LinuxHostName
+            $User = "root"
+            $Port = 5986
+            try
+            {
+                winrm enumerate http://schemas.microsoft.com/wbem/wscim/1/cim-schema/2/OMI_Identify?__cimnamespace=root/omi -r:https://${hostname}:$Port -auth:Basic -u:$User -p:$badPassword -skipcncheck -skipcacheck -encoding:utf-8
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | Should be "NativeCommandError"
+            }
+        }
+        It "Remoting from Mac OS to Linux with basic authentication should work" -Skip:($IsLinux -Or $IsWindows){
+            $hostname = $LinuxHostName
+            $User = "root"
+            $PWord = convertto-securestring $passwordString -asplaintext -force
+            $cred = New-Object �TypeName System.Management.Automation.PSCredential �ArgumentList $User, $PWord
+            $sessionOption = New-PSSessionOption -SkipCACheck -SkipRevocationCheck -SkipCNCheck
+            $mySession = New-PSSession -ComputerName $hostname -Credential $cred -Authentication Basic -UseSSL -SessionOption $sessionOption
+            $result = Invoke-Command -Session $mySession {Get-Host}
+            $result|Should Not BeNullOrEmpty
+        }
+        It "Remoting from Mac OS to Linux with basic authentication by winrm enumerate OMI_Identify should work" -Skip:($IsLinux -Or $IsWindows){
+            $hostname = $LinuxHostName
+            $User = "root"
+            $Port = 5986
+            $result = winrm enumerate http://schemas.microsoft.com/wbem/wscim/1/cim-schema/2/OMI_Identify?__cimnamespace=root/omi -r:https://${hostname}:$Port -auth:Basic -u:$User -p:$passwordString -skipcncheck -skipcacheck -encoding:utf-8
+            $result|Should Not BeNullOrEmpty
+        }
+        It "Remoting from Linux to Windows with basic authentication should work" -Skip:($IsOSX -Or $IsWindows){
+            $hostname = $WindowsHostName
+            $User = "scx\scxsvc"

```

Comment:
The user name should not be hard-coded. People with different user names might still want to run the tests. Consider using an environment variable like the other parameters. Fix on the different tests.